### PR TITLE
Initial Dockerfile

### LINKED ISSUE #193: create and pulish docker image
## Scope
- create a docker image

## Deliverables
- docker image commit
- registered organization on docker.com
- updated readme on docs.leapdao.org

## Gain for the project
- content for social 

## Roles: 
~~bounty gardener: @pinkiebell / 10%~~
bounty gardener/worker: @pinkiebell  / 70% 
bounty reviewer: @eezcjkr  / 20% | @troggy / 10%

### PR DESCRIPTION
**Pull request by pinkiebell**
~Depends on #137~

First image 🎉 
https://quay.io/repository/leapdao/leap-node

Fixes #193 (except for documentation)

### CODE REVIEW

**Comment by eezcjkr** [comment]
So if I understand correctly this one also addresses https://github.com/leapdao/leap-node/issues/111 ?

**Comment by pinkiebell** [comment]
> So if I understand correctly this one also addresses #111 ?

It paves the way, I did not tried an arm build. If the dependencies support that, it should work out of the box (building the docker image on an arm machine).

File: lotion/index.js [hunk 1]
```diff
@@ -198,16 +196,7 @@ function Lotion(opts = {}) {
...
         await tendermint.synced;
 
-        // serve genesis.json and get GCI
-        const genesisJson = fs.readFileSync(
-          join(lotionPath, 'config', 'genesis.json'),
-          'utf8'
-        );
-        const { GCI } = serveGenesisGCI(genesisJson);
-
-        announceSelfAsFullNode({ GCI, tendermintPort });
```
[the remark this answers is not in the document]
Comment (pinkiebell) [reply lotion/index.js]: Found no regressions, but I also had problems with integration tests timing out randomly over the place. But that is not related to this PR 🤷‍♂️

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+WORKDIR /opt/leap-node
+RUN apk add --no-cache --update --virtual build_deps $BUILD_DEPS
+COPY . /opt/leap-node
+RUN chmod 755 bin.js
+RUN npm install --production --no-package-lock yarn
+RUN ./node_modules/.bin/yarn install --production
+RUN ./node_modules/.bin/yarn link
+RUN apk del build_deps
+CMD ["leap-node", \
+  "--config=http://node1.testnet.leapdao.org:8645", \
```
Comment (troggy) [inline Dockerfile]: I suggest to take this from env ARG, so we can specify which network we want connect to

Comment (troggy) [reply Dockerfile]: default may be `https://testnet-node1.leapdao.org` (port 46691 is opened now)

**Comment by pinkiebell** [comment]
@troggy @eezcjkr 
I introduced support for environment variables for some cli args, and I added leap-{testnet,mainnet}.json presets.

**Comment by troggy** [comment]
optimisitically merging

**Pull request merged**